File: from-1401-to-1500/1422-maximum-score-after-splitting-a-string/solution.go
func maxScore(s string) int {
    nOnes := 0
    for i := 0; i < len(s); i++ {
        if s[i] == '1' {
            nOnes += 1
        }
    }

    score, maxScore := nOnes, 0
    for i := 0; i < len(s)-1; i++ {
        if s[i] == '0' {
            score += 1
        } else {
            score -= 1
        }
        if score > maxScore {
            maxScore = score
        }
    }
    return maxScore
}

File: from-1401-to-1500/1422-maximum-score-after-splitting-a-string/solution.py
class Solution:
    def maxScore(self, s: str) -> int:
        nOnes = sum(1 if c == '1' else 0 for c in s)
        
        score, maxScore = nOnes, 0
        for c in s[:-1]:
            if c == '0':
                score += 1
            else:
                score -= 1
                
            maxScore = max(score, maxScore)

        return maxScore